services/provider_credential_service.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "./services"
+require_relative "../shared/logging_module"
+
+module FastlaneCI
+  # Provides access to provider credential related methods
+  class ProviderCredentialService
```

Comment:
I'm not sold on the idea that `ProviderCredentialService` should be separate from the `UserService`. I think because `User` owns `ProviderCredential`s, it might make sense to continue to manage that in the actual `UserService`. 
What do you think?

Reply:
Hmm. That's a good point. My intention was to keep them separate for sake of Single Responsibility Principle (`UserService` is only responsible for users, `ProviderCredentialService` is only responsible for provider credentials), but that's a good point about the `User` owning the `ProviderCredential`. If you want, I can move these two methods to the `UserService`.

Reply:
I like single responsible principle (in principle 🙃), but because users are ultimately the holders of credentials, I think it makes the most sense to bundle it together.

Reply:
Sounds good to me! :)